sklearn/mixture/_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,416 @@
+
+    max_components : int or None, default=10
+        The maximum number of mixture components to consider.
+        Must be greater than or equal to ``min_components``.
+
+    covariance_type : {'full', 'tied', 'diag', 'spherical', 'all' (default)},
+            optional
+        String or list/array describing the type of covariance parameters
+        to use.
+        If a string, it must be one of:
+
+        - 'full'
+            each component has its own general covariance matrix
+        - 'tied'
+            all components share the same general covariance matrix
+        - 'diag'
+            each component has its own diagonal covariance matrix
+        - 'spherical'
+            each component has its own single variance
+        - 'all'
+            considers all covariance structures in
+            ['spherical', 'diag', 'tied', 'full']
+
+        If a list/array, it must be a list/array of strings containing only
+        'spherical', 'tied', 'diag', and/or 'spherical'.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        There is randomness in k-means initialization of
+        :class:`sklearn.mixture.GaussianMixture`. This parameter is passed to
+        :class:`~sklearn.mixture.GaussianMixture` to control the random state.
+        If int, ``random_state`` is used as the random number generator seed.
+        If RandomState instance, ``random_state`` is the random number
+        generator; If None, the random number generator is the
+        RandomState instance used by ``np.random``.
+
+    n_init : int, optional (default = 1)
+        If ``n_init`` is larger than 1, additional
+        ``n_init``-1 runs of :class:`sklearn.mixture.GaussianMixture`
+        initialized with k-means will be performed
+        for all covariance parameters in ``covariance_type``.
+
+    init_params : {‘kmeans’ (default), ‘k-means++’, ‘random’, ‘random_from_data’}
+        The method used to initialize the weights, the means and the precisions
+        for Gaussian mixture modeling.
+
+    max_iter : int, optional (default = 100)
+        The maximum number of EM iterations to perform.
+
+    verbose : int, optional (default = 0)
+        Enable verbose output. If 1 then it prints the current initialization
+        and each iteration step. If greater than 1 then it prints also
+        the log probability and the time needed for each step.
+
+    criterion : str {"bic" or "aic"}, optional, (default = "bic")
+        Select the best model based on Bayesian Information Criterion (bic) or
+        Aikake Information Criterion (aic).
+
+
+    Attributes
+    ----------
+    best_criterion_ : float
+        The best (lowest) Bayesian or Aikake Information Criterion.
+
+    n_components_ : int
+        Number of clusters for the model with the best bic/aic.
+
+    covariance_type_ : str
+        Covariance type for the model with the best bic/aic.
+
+    best_model_ : :class:`sklearn.mixture.GaussianMixture`
```

Comment:
in lassolarsIC, there is no "sub-object" with the best model; rather the whole class just operates as if it is that model. does that make sense? while i cant speak for them, my guess is this is closer to what they'd be expecting

Reply:
I add the attributes like `weights_`, `means_` from GaussianMixture into GaussianMixtureIC, but I found that I still need to save the best model (I call `best_estimator_` in the newest version) in order to all `predict`. Did I understand you correctly?